docs/api/service_container.md

```diff
@@ -0,0 +1,32 @@
+# Service container
+
+The service container is a special object that facilitates dependency resolution.
+It is based on the [dependency injection design pattern](http://en.wikipedia.org/wiki/Dependency_injection).
+
+Dependency injection injects all needed objects and configuration into your business logic objects (known as services).
+It avoids the massive use of singletons, global variables or complicated factories and makes code much more readable and testable.
+
+The main issue with this pattern is how to resolve the dependencies for your services.
+This is where the service container comes into play. The role of a service container is to build and maintain your services and their dependencies.
+Each time you need a service, you may ask the service container for it.
+It either builds the service with the configuration you provided, or give you an existing instance if it is already available.
+
+[[= product_name =]] uses the [Symfony service container](http://symfony.com/doc/current/service_container.html).
+
+!!! tip
+
+    To learn more about the service container, see the documentation for the [Symfony DependencyInjection component](http://symfony.com/doc/current/components/dependency_injection.html) documentation.
+
+## Service tags
+
+Service tags used by the Symfony service container are a way to dedicate services to a specific purpose. They are usually used for extension points.
+
+For example, if you want to register a [Twig extension](http://twig.sensiolabs.org/doc/advanced.html#creating-extensions) to add custom filters,
```

Comment:
This link seems to be broken, I only managed to find this https://symfony.com/doc/current/templating/twig_extension.html but I'm not sure if it covers the topic at all.

Reply:
It does, actually :+1:

Reply:
```suggestion
For example, if you want to register a [Twig extension](https://symfony.com/doc/current/templating/twig_extension.html) to add custom filters,
```